pkg/portfwd/client.go

```diff
@@ -29,14 +29,14 @@ func HandleTCPConnection(ctx context.Context, client *guestagentclient.GuestAgen
 	rw := &GrpcClientRW{stream: stream, id: id, addr: guestAddr}
 	g.Go(func() error {
 		_, err := io.Copy(rw, conn)
-		if errors.Is(err, io.EOF) {
+		if err != nil && errors.Is(err, io.EOF) {
 			return nil
 		}
 		return err
```

Comment:
We can simplify this further by completely removing `errors.Is(err, io.EOF)`:
```suggestion
		return err
```

According [to the doc](https://pkg.go.dev/io#Copy), `io.Copy` never returns `io.EOF`:

> Copy copies from src to dst until either EOF is reached on src or an error occurs. It returns the number of bytes copied and the first error encountered while copying, if any.
>
> A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported.

Reply:
Yes, the check was removed now.